pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -151,6 +154,24 @@ cdef dict timedelta_abbrevs = {
 
 _no_input = object()
 
+# pytimedeltas may fall in the following range: [-999999999 days, 1000000000 days)
+# for example, 999999999 days, 86399 seconds is in the range,
+# whereas -999999999 days, -86399 seconds is not in the range
+
+# the relevant constant is defined in CPython _datetimemodule.c,
+# rather than datetime.h, so we have to define it again here
+cdef int64_t MAX_DELTA_DAYS = 999999999
```

Comment:
OK so I see these are located in "_datetimemodule.c" so its not part of the CPython API. I don't think we should be doing this at all then, as we are mixing into CPython internals.

It looks like CPython anyways already does bounds checking when creating a timedelta, so we should just rely on that. 

https://github.com/python/cpython/blob/d4cea794a7b9b745817d2bd982d35412aef04710/Modules/_datetimemodule.c#L1101

Reply:
Makes sense. The key here is to avoid breaking cases in which pd.Timedelta allows for values outside the bounds of pytimedelta, as referenced by the original code comment. I’m thinking we try to pass the value, and pass zero upon an OverflowError from CPython (possibly raising a warning). Is that along the lines of what you're thinking?

Reply:
Loosely. I'm still probably missing the bigger picture, but I think at least catching the OverflowError is headed in the right direction. So worth trying that instead and see where that takes us